internal/k8s/reconciler/http_route_test.go

```diff
@@ -26,7 +27,7 @@ func TestConvertHTTPRoute(t *testing.T) {
 	headerMatchType := gw.HeaderMatchExact
 	weight := int32(10)
 	protocol := "https"
-	hostname := gw.Hostname("example.com")
+	hostname := gw.PreciseHostname("example.com")
```

Comment:
This requirement was introduced by https://github.com/kubernetes-sigs/gateway-api/pull/956

Reply:
I have a love-hate relationship with all that type aliasing...